frontend/packages/pipelines-plugin/src/components/repository/__tests__/RepositoryLinkList.spec.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { ResourceLink, ExternalLink, CopyToClipboard } from '@console/internal/components/utils';
+import { PipeLineRunWithRepoMetadata } from '../../../test-data/pipeline-data';
+import RepositoryLinkList from '../RepositoryLinkList';
+
+jest.mock('react-i18next', () => {
+  const reactI18next = require.requireActual('react-i18next');
+  return {
+    ...reactI18next,
+    useTranslation: () => ({ t: (key: string) => key }),
+  };
+});
+
+describe('RepositoryLinkList', () => {
+  it('should not render dl when no repository related labels exists', () => {
+    const repositoryWrapper = shallow(
+      <RepositoryLinkList pipelineRun={PipeLineRunWithRepoMetadata[0]} />,
+    );
+    expect(repositoryWrapper.find('dl').exists()).toBe(false);
+  });
+
+  it('should return correct details when repository related labels exists', () => {
```

Comment:
I feel like we shouldn't write tests like this... Digging into components to see if a passed prop is equal to the passed value is giving us what exactly? I feel like this is just saying "we wrote that line of code to do that static passing". 

If you're concerned about a computation or logical split of code out of something, that's where utilities and hooks come into play -- test THOSE.

Also primarily we should only be testing conditional renderings... static renderings don't give us a lot. If you need to select a component to see if something is inside it, add a `data-test` attribute so it can also be used by functional tests. We should avoid selecting `2nd div, is span inside it?` kinda tests.

Reply:
@andrewballantyne 
It's more around tracking that SHA is wrapped in CopyToClipboard and URL is exposed as an externalLink and Repo as a resourceLink. Also logically it checks the number of Repository related fields rendered which are conditional.

Let me whether we should keep these or remove

Reply:
I've asked @karthikjeeyar to follow up with you. Look out for his message or reach out to him first.